packages/client/hmi-client/src/services/dataset.ts

```diff
@@ -50,6 +50,25 @@ async function getClimateDataset(datasetId: string): Promise<Dataset | null> {
 	return response?.data ?? null;
 }
 
+async function getClimateSubsetId(
+	esgfId: string,
+	parentDatasetId: string,
+	envelope: string,
+	timestamps?: string,
+	thinFactor?: string
+): Promise<string | null> {
+	const url = `/climatedata/queries/subset-esgf/${esgfId}?parent-dataset-id=${parentDatasetId}&envelope=${envelope}`;
+	if (timestamps) url.concat(`&timestamps=${timestamps}`);
+	if (thinFactor) url.concat(`&thin-factor=${thinFactor}`);
+
+	const response = await API.get(url).catch((error) => {
+		logger.error(
+			`Error: climate data service was not able to retrieve the subset of the dataset ${esgfId} ${error}`
+		);
+	});
```

Comment:
Errors are handled by `api/api.ts` and displays whatever the controllers sends back.
Here you can only handle `2XX` HTTP message. So instead test that is not a 200.
```suggestion
	const response = await API.get(url);
	if (response.status !== 200 || isEmtpy(response.data)) {
		logger.error(`Climate-data service was not able to retrieve the subset of the dataset ${esgfId}`, error);
		return null
	}
```

Reply:
I vote on the original code there - `response.status` can _only_ be a 2XX in this case because if it was a 4XX or 5XX error `api.ts` would catch the error, so this new setup wouldn't do much.

With the catch on the error like how it is it matches how we do it everywhere else, and deals with the off chance that api.ts runs into an issue.

Reply:
The new code catch non 200 status, like a `204 No Content`, or `202 Accepted`, or that the response.data is null.